test/browser/suspense.test.js

```diff
@@ -0,0 +1,235 @@
+/*eslint-env browser, mocha */
+import { setupRerender } from 'preact/test-utils';
+import { expect } from 'chai';
+import { createElement as h, render, Component, Suspense, lazy } from '../../src/index';
+import { setupScratch, teardown } from '../_util/helpers';
+
+class LazyComp extends Component {
+	render() {
+		return <div>Hello Lazy</div>;
+	}
+}
+
+class CustomSuspense extends Component {
+	constructor(props) {
+		super(props);
+		this.state = { done: false };
+	}
+	render() {
+		if (!this.state.done) {
+			throw new Promise((res) => {
+				setImmediate(() => {
+					this.setState({ done: true });
+					res();
+				});
+			});
+		}
+
+		return (
+			<div>
+				Hello CustomSuspense
+			</div>
+		);
+	}
+}
+
+class ThrowingSuspense extends Component {
+	constructor(props) {
+		super(props);
+		this.state = { done: false };
+	}
+	render() {
+		if (!this.state.done) {
+			throw new Promise((res, rej) => {
+				setImmediate(() => {
+					this.setState({ done: true });
+					rej(new Error('I\'m a throwing Suspense!'));
+				});
+			});
+		}
+
+		return (
+			<div>
+				Hello ThrowingSuspense
+			</div>
+		);
+	}
+}
+
+class Catcher extends Component {
+	constructor(props) {
+		super(props);
+		this.state = { error: null };
+	}
+
+	componentDidCatch(e) {
+		this.setState({ error: e });
+	}
+
+	render() {
+		return this.state.error ? (
+			<div>
+				Catcher did catch: {this.state.error.message}
+			</div>
+		) : this.props.children;
+	}
+}
+
+const Lazy = lazy(() => new Promise((res) => {
+	setImmediate(() => {
```

Comment:
`setImmediate` is a [non-standard API](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate) and only supported in IE

Reply:
Replaced with `setTimeout`. Thx for the heads up :)